### PR TITLE
Asset-hub-royalties

### LINKED ISSUE #7733: Royalties on Asset Hub
It would be cool to have creators' royalties on AssetHub Kusama and AssetHub Polkadot.

Can anyone add it? 🔥🔥

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #7733
- [x] Requires deployment - stick
- [x] won't work until #7771 is merged
* https://github.com/kodadot/stick/pull/132

#### Did your issue had any of the "$" label on it?
$

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

## Related transacrions

1. tx with set attributes (on chain royalty)
https://assethub-kusama.subscan.io/extrinsic/5640724-2
2. buy single item with royalty
https://assethub-kusama.subscan.io/extrinsic/5640390-2
3. buy 3 items with royalties (shopping cart)
https://assethub-kusama.subscan.io/extrinsic/5645506-2

## items minted with royalty:
/ahk/gallery/12-10
/ahk/gallery/12-11
/ahk/gallery/12-12

- [ ] My fix has changed UI

### <samp>🤖 Generated by Copilot at b6f31df</samp>

This pull request adds and improves the functionality for handling royalties on different chains and NFT standards. It enables minting and paying royalties on Statemine, toggles Rmrk-specific features based on the chain, and refactors the code for verifying and normalizing royalty data. It affects the files `CreateNft.vue`, `transactionMintStatemine.ts`, `transactionBuy.ts`, `useIsChain.ts`, `support.ts`, and `utils.ts`.

### <samp>🤖 Generated by Copilot at b6f31df</samp>

> _Sing, O Muse, of the crafty developers who wrought_
> _New features for the NFT marketplace, and sought_
> _To honor the creators with `royaltyFee` and `payTips`_
> _And to validate their data with `verifyRoyalty` and `isRmrk` scripts._

### CODE REVIEW

File: utils/support.ts [hunk 1]
```diff
@@ -46,12 +46,15 @@ export const somePercentFromTX = (api: ApiPromise, price: number | string) => {
   return asBalanceTransfer(api, resolveSupportAddress(api), fee)
 }
 
+export const royaltyFee = (price: string | number, royaltyPercent: number) =>
+  (Number(price) * royaltyPercent) / 100
```
Comment (roiLeo) [inline utils/support.ts:50]: looks different as before, have you check if it doesn't break rmrk royalty?

**Review by roiLeo — APPROVED** [review]
otherwise code lgtm

File: composables/useIsChain.ts [hunk 2]
```diff
@@ -8,6 +8,9 @@ export default function (prefix: ComputedRef<Prefix>) {
   const isRemark = computed(
     () => prefix.value === 'rmrk' || prefix.value === 'ksm',
   )
+
+  const isRmrk = computed(() => prefix.value === 'rmrk')
+
```
Comment (vikiival) [inline composables/useIsChain.ts:13]: what's the diff?

File: composables/transaction/transactionBuy.ts [hunk 3]
```diff
@@ -6,9 +6,35 @@ import {
...
+
+  return legacy
+    ? payRoyaltyTx(api, price, normalizedRoyalty)
+    : api.tx.nfts.payTips([
+        {
+          collection: collectionId,
+          item: tokenId,
+          receiver: normalizedRoyalty.address,
+          amount: royaltyFee(price, normalizedRoyalty.amount),
+        },
```
Comment (vikiival) [inline composables/transaction/transactionBuy.ts:35]: move to utils?

**Review by vikiival — COMMENTED** [review]
~

File: composables/useIsChain.ts (hunk 2 above)
Comment (daiagi) [reply composables/useIsChain.ts:13]: between?

royalty amount input should appear for all except rmrk

File: composables/transaction/transactionBuy.ts (hunk 3 above)
Comment (daiagi) [reply composables/transaction/transactionBuy.ts:35]: i don't see the value in it
you will have a function with 4 params plus API

why not just use `api.tx.nfts.payTips` directly?

File: utils/support.ts (hunk 1 above)
Comment (roiLeo) [reply utils/support.ts]: 83qxa3

**Comment by daiagi** [comment]
@prury pleae note that royalty can only be whole numbers (integers) until new squid is released
@vikiival

**Comment by vikiival** [comment]
> @prury pleae note that royalty can only be whole numbers (integers) until new squid is released
> @vikiival

It is posible to do that automatized way 

https://github.com/kodadot/stick/compare/release-squid...main

**Comment by daiagi** [comment]
> It is posible to do that automatized way

thought so but didn't want to step in and do it without your consent

**Comment by AshutoshSingh72** [comment]
@vikiival what ? He is adding royalty to each nft unless of adding into the collection

**Comment by vikiival** [comment]
> @vikiival what ? He is adding royalty to each nft unless of adding into the collection

Sir I do not understand what you mean.
can you please rephrase?

File: tests/e2e/footer.spec.ts [hunk 4]
```diff
@@ -68,10 +68,10 @@ const footerSocialMediaLinks = [
...
     linkAddress: 'https://www.youtube.com/channel/UCEULduld5NrqOL49k1KVjoA/',
   },
-  {
-    linkName: 'Instagram',
-    linkAddress: 'https://www.instagram.com/kodadot.xyz/',
-  },
+  // {
+  //   linkName: 'Instagram',
+  //   linkAddress: 'https://www.instagram.com/kodadot.xyz/',
+  // },
```
Comment (roiLeo) [inline tests/e2e/footer.spec.ts:74]: let's fix it in another PR or maybe @prury has an idea?

Comment (prury) [reply tests/e2e/footer.spec.ts:74]: > let's fix it in another PR or maybe @prury has an idea?

can we leave it like that and I'll apply a proper fix in another PR?

**Comment by yangwao** [comment]
Thanks!
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 4.34 USD/DOT ~ 11.521 $DOT
🧗 [1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8](https://polkadot.subscan.io/account/1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)
🔗 [0x05c0694ad469c351f39e4db307e6bb0f1819d40bb35625e221b4fd6a41e24429](https://polkadot.subscan.io/extrinsic/0x05c0694ad469c351f39e4db307e6bb0f1819d40bb35625e221b4fd6a41e24429)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**